coalib/results/Diff.py

```diff
@@ -70,6 +71,23 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unidiff(cls, diff_array):
+        """
+        Creates a Diff object from given unified diff.
+        :param diff_array: list of lines of unified diff.
+        """
+        source_array = []
+        target_array = []
+        patch_set = unidiff.PatchSet(diff_array)
+        for patched_file in patch_set:
+            for hunk in patched_file:
+                for line in hunk.source_lines():
+                    source_array.append(str(line)[1:])
```

Comment:
> Not necessarily, I guess it depends on the unified diff itself. But why do we need complete files to generate the Diff object?

Because thats how it  is implemented.. It takes in the whole file in `self._file` and stores the modified one in `self.modified`, and the file is modified with calls to `delete_lines`, `modify_lines`, `change_line` mutators AFAIU. You have to atleast generate the modified file completely from the diff.. or come up with better implementation ideas IMO